corehq/apps/domain/views/releases.py

```diff
@@ -108,58 +109,63 @@ class ManageReleasesByAppProfile(BaseProjectSettingsView):
     page_title = ugettext_lazy("Manage Releases By App Profile")
 
     @cached_property
-    def form(self):
-        return ManageReleasesByAppProfileForm(
+    def creation_form(self):
+        return CreateManageReleasesByAppProfileForm(
             self.request,
             self.domain,
             data=self.request.POST if self.request.method == "POST" else None,
         )
 
     @staticmethod
-    def _get_initial_app_profile_details(domain, version, app_id, build_profile_id):
+    def _get_initial_app_build_profile_details(build_profiles_per_app, app_id, app_build_profile_id):
         # only need to set when performing search to populate with initial values in view
-        if app_id and version:
-            build_doc = get_build_doc_by_version(domain, app_id, version)
-            if build_doc:
-                return [{
-                    'id': _id,
-                    'text': details['name'],
-                    'selected': build_profile_id == _id
-                } for _id, details in build_doc['build_profiles'].items()]
+        if build_profiles_per_app and app_id and app_id in build_profiles_per_app:
```

Comment:
I think you can leave off the `and app_id`, assuming there wouldn't be a `None` in `build_profiles_per_app`.

Reply:
it's coming from search and is a required field in the form, so ideally it should always be present but good to still check it.